src/Symfony/Component/Security/Http/Firewall/AbstractPreAuthenticatedListener.php

```diff
@@ -54,6 +54,15 @@ public function __construct(TokenStorageInterface $tokenStorage, AuthenticationM
      */
     final public function handle(GetResponseEvent $event)
     {
+        $token = $this->tokenStorage->getToken();
+        if(null !== $token && false === ($token instanceof PreAuthenticatedToken) && $token->isAuthenticated()) {
+            if (null !== $this->logger) {
+                $this->logger->debug('An authenticated token is already provided, bypass PreAuthenticatedListener.', array('token' => (string) $this->tokenStorage->getToken()));
```

Comment:
I'd personally prefer to see the to string method being called here in case of refactoring one day